doc/features/mapper/queries/README.md

```diff
@@ -0,0 +1,294 @@
+
+When a model is mapped to multiple tables, it will update a row in each table when all the primary keys are specified
+ grouped in a logged batch (either all or none of the update operations will succeed).
+
+Additionally, `update()` supports conditional clause for [lightweight transactions (CAS)][lwt] that allows to 
+specify the condition that has to be met for the update to occur. Please note that using IF conditions will incur a 
+non-negligible performance cost on the server-side so this should be used sparingly.
+
+### Usage examples
+
+#### Update a single object into one or more tables
+
+Update a video
+
+```javascript
+await videoMapper.update({ videoId, name, addedDate, userId, description });
+```
+
+#### Update few selected fields of an object
+
+Update only the name and the added date, regardless of the other properties specified in the object.
+
+```javascript
+await videoMapper.update(video, { fields: ['videoId', 'name', 'description'] });
+```
+
+#### Update an object using a [conditional statement][lwt]
+
+Update a video when the existing name contains a certain value.
+
+```javascript
+await videoMapper.update({ videoId, name, description }, { when: { name: 'original name' } });
+```
+
+## Delete
+
+Use the `remove()` method on a `ModelMapper` instance to delete an object.
+
+When a model is mapped to multiple tables, it will delete the row on each table when all the primary keys are specified
+ grouped in a logged batch (either all or none of the delete operations will succeed).
+
+Additionally, `remove()` supports conditional clause for [lightweight transactions (CAS)][lwt] that allows to 
+specify the condition that has to be met for the delete to occur. Please note that using IF conditions will incur a 
+non-negligible performance cost on the server-side so this should be used sparingly.
+
+### Usage examples
+
+#### Delete a single object into one or more tables
+
+Delete a video
+
+```javascript
+await videoMapper.delete({ videoId });
+```
+
+#### Delete an object using a [conditional statement][lwt]
+
+Delete a video when the existing name contains a certain value.
+
+```javascript
+await videoMapper.delete({ videoId }, { when: { name: 'original name' } });
+```
+
+## Group mutations in a batch
+
+You can batch multiple operations for both single partition and multiple partitions when [atomicity][atomicity] and
+[isolation][isolation] is a requirement for a group of changes.
+
+You can use the field `batching` of a `ModelMapper` to create each item of a batch and the `batch()` method of the 
+`Mapper` to submit the request.
+
+### Usage example
+
+#### Update a group of objects
+
+Update two videos from a user in a batch.
+
+```javascript
+const changes = [
+  videoMapper.batching.update({ userId, videoId1, name1, addedDate1 }),
+  videoMapper.batching.update({ userId, videoId2, name2, addedDate2 })
+];
+
+// Execute the batch
+await mapper.batch(changes);
```

Comment:
One thing I find odd with this:  You can do operations on the ModelMapper (such as `find`, `update`, `delete`), but in order to execute a batch, you need to use `Mapper`.  For convenience, is there any reason why we couldn't add a `batch` method to `ModelMapper`?  This way you don't need the `Mapper` reference to do a `batch`.

Reply:
I've placed inside `Mapper` because it might contain queries for different models/tables:

```javascript
const changes = [
  videoMapper.batching.update({ userId, videoId1, name1, addedDate1 }),
  userStatusMapper.batching.update({ userId, status: 'User updated video' })
];

// Execute the batch
await mapper.batch(changes);
```

Reply:
Yeah that does make sense, but if you are doing batch updates on a single table (which is pretty common), that means you have to have references to both a `ModelMapper` and `Mapper` instances to execute that operation, where with other operations you only need `ModelMapper`.

I'm not sure how important it is, but I can see it being convenient having a `ModelMapper.batch` that only works if the batch only contains updates for that model.